infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/logreader/LogicalGroupLogEntryReader.java

```diff
@@ -12,10 +34,98 @@
+        Set<UUID> txEntryStreamIds = new HashSet<>(entry.getEntries().keySet());
+
+        if (txEntryStreamIds.contains(CLIENT_CONFIG_TABLE_ID) &&
+                isCurrentSessionImpacted(entry.getEntries().get(CLIENT_CONFIG_TABLE_ID))) {
+            log.info("Group destination change detected, log entry sync will be stopped and a forced snapshot " +
+                    "sync will be triggered！");
+            throw new GroupDestinationChangeException();
+        }
+
+        return super.isValidTransactionEntry(entry);
+    }
+
+    /**
+     * Helper method for checking if the smr entries present in the opaque stream (that current session's log entry
+     * reader is tracking) contain any group destination change with respect to current session. The opaque entries
+     * will be deserialized to verify if the Sink cluster of current session is among the target destinations.
+     *
+     * @param groupConfigTableEntries SMREntries of LogReplicationModelMetadataTable.
+     * @return True if current session is impacted by group destination config change, false otherwise.
+     */
+    private boolean isCurrentSessionImpacted(List<SMREntry> groupConfigTableEntries) {
+        Set<String> groups = ((LogReplicationLogicalGroupConfig) replicationContext.getConfig(session))
+                .getLogicalGroupToStreams().keySet();
+
+        for (SMREntry smrEntry : groupConfigTableEntries) {
+            // Get serialized form of arguments for registry table. They were sent in OpaqueEntry and
+            // need to be deserialized using ProtobufSerializer
+            Object[] objs = smrEntry.getSMRArguments();
+            ByteBuf keyBuf = Unpooled.wrappedBuffer((byte[]) objs[0]);
+            ClientDestinationInfoKey clientInfoKey = (ClientDestinationInfoKey) protobufSerializer
+                    .deserialize(keyBuf, null);
+
+            if (clientInfoKey.getModel() != session.getSubscriber().getModel() ||
+                    !clientInfoKey.getClientName().equals(session.getSubscriber().getClientName())) {
+                // SMREntry for other clients
+                continue;
+            }
+
+            ByteBuf valueBuf = Unpooled.wrappedBuffer((byte[]) objs[1]);
+            CorfuRecord<DestinationInfoVal, Message> destinations =
+                    (CorfuRecord<DestinationInfoVal, Message>) protobufSerializer.deserialize(valueBuf, null);
+
+            boolean isGroupAdded = !groups.contains(clientInfoKey.getGroupName()) &&
+                    destinations.getPayload().getDestinationIdsList().contains(session.getSinkClusterId());
+            boolean isGroupRemoved = groups.contains(clientInfoKey.getGroupName()) &&
+                    !destinations.getPayload().getDestinationIdsList().contains(session.getSinkClusterId());
+
+
+            if (isGroupAdded || isGroupRemoved) {
+                if (isGroupAdded) {
+                    log.info("New group added for the Sink of current session. Group=[{}], Sinks=[{}]",
```

Comment:
minor: the 2 cases can be combined into a single log statement according whether a group was added or removed.

Reply:
I was trying to log different msgs based on added or removed. Do you mean just log the group name and destinations?

Reply:
no, the string 'added' or 'removed' can be according to the condition.   For example:
`isGroupAdded ? log.info(".....") : log.info("....)`

At any time, only 1 case will be true(added or removed).